### PR TITLE
*: Don't perform address normalization locally

### PR DESCRIPTION
**Pull request by brancz**
Instead send the necessary information to perform normalization to the server and let the server do it. This also allows the server to potentially retrieve other sources for normalization calculations.

Client side to https://github.com/parca-dev/parca/pull/3625

### CODE REVIEW

**Review by kakkoyun — APPROVED** [review]
Awesome 🏆 🥇 💯

File: pkg/runtime/lua.go [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+	"errors"
+	"fmt"
+)
+
+var luaJitIdentifyingSymbols = [][]byte{
+	[]byte("luaJIT"),
+}
+
+// HACK: This is a somewhat a brittle check.
+func IsLuaJit(f *elf.File) (bool, error) {
```
Comment (kakkoyun) [inline pkg/runtime/lua.go:27]: It doesn't seem to be related to PR. Maybe a leftover?

File: pkg/process/maps.go [hunk 2]
```diff
@@ -318,65 +315,81 @@ func (m *Mapping) findProgramHeader(ef *elf.File, addr uint64) (*elf.ProgHeader,
...
+		return nil, fmt.Errorf("specified address %x is outside the mapping range [%x, %x]", addr, m.StartAddr, m.EndAddr)
 	}
 
 	loadSegment, err := m.findProgramHeader(ef, addr)
 	if err != nil {
-		return 0, fmt.Errorf("failed to find program header for mapping %#v: %w", m, err)
+		return nil, fmt.Errorf("failed to find program header for mapping %#v: %w", m, err)
 	}
 
-	base, err := elfreader.Base(
```
Comment (kakkoyun) [inline pkg/process/maps.go:334]: Since we move this method to upstream Parca, shall we clean it up? Is it used anywhere else?

File: pkg/runtime/lua.go (hunk 1 above)
Comment (brancz) [reply pkg/runtime/lua.go:27]: whops, yes, thanks

File: pkg/process/maps.go (hunk 2 above)
Comment (brancz) [reply pkg/process/maps.go:334]: no you're right!
